lib/Helper/ImageService/ImageFileHelper.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace OCA\Cookbook\Helper\ImageService;
+
+use OCP\IL10N;
+use OCP\Files\File;
+use OCP\Files\Folder;
+use OCP\Files\NotFoundException;
+use OCA\Cookbook\Exception\RecipeImageExistsException;
+
+class ImageFileHelper {
+	private const NAME_MAIN = ImageSize::NAME_MAP[ImageSize::PRIMARY_IMAGE];
+
+	/**
+	 * @var IL10N
+	 */
+	private $l;
+
+	public function __construct(
+		IL10N $l
+		) {
+		$this->l = $l;
+	}
+
+	/**
+	 * Get the main image from a recipe
+	 *
+	 * @param Folder $recipeFolder The folder of the recipe
+	 * @return File The image file
+	 * @throws NotFoundException if no image has been found
+	 * @todo Do not use low-level functions of NC core here.
+	 */
+	public function getImage(Folder $recipeFolder): File {
+		return $recipeFolder->get(self::NAME_MAIN);
```

Comment:
The `File` is a direct subclass of `Node`. So, yes, there is the theoretical option that someone creates a folder `full.jpg` which will break this code.
I was thinking (and starting to implement) an abstraction layer of the File/Folder/Node interface to be more tailored towards our needs. Just an additional layer (or two). Maybe there we can handle that case better. I suspect there are multiple cases where this might happen (we should definitively start to use psalm) and that needs addressing.